Source/Dafny/Resolver.cs

```diff
@@ -11075,9 +11075,14 @@ public class ResolveTypeReturn {
             s.TargetStmt = target;
           }
         } else {
-          if (loopStack.Count < s.BreakCount) {
-            var jump = s.IsContinue && s.BreakCount != 1 ? "break/continue" : s.Kind;
-            reporter.Error(MessageSource.Resolver, s, $"trying to {jump} out of more loop levels than there are enclosing loops");
+          var jumpStmt = s.BreakCount == 1 ?
+            $"a non-labeled '{s.Kind}' statement" :
+            $"a '{Util.Repeat(s.BreakCount - 1, "break ")}{s.Kind}' statement";
```

Comment:
s.BreakCount might be zero at this stage, which could trigger an infinite loop. Consider either testing it or changing Repeat so that 0 or -1 are considered the same.

Reply:
I disagree. There's an invariant in class `BreakStmt` that says

```
Contract.Invariant(TargetLabel != null || 1 <= BreakCount);
```

and the `if` statement surrounding the assignment to `jumpStmt` ensures that this code is used only when `s.TargetLabel == null`.

The count `s.BreakCount` include the final `continue`. So, the name of this field is not the best. Would `BreakContinueCount` or `Levels` be more readily understood?

Reply:
Good argument. I can't see the surrounding block in GitHub because the file is too large, so I couldn't have had this assumption, I'm just watching over your shoulder :-)
However, while debugging Dafny, I have seen surprisingly many times invariants which did not hold but the code did not crash. It looks like the invariants are not checked (how odd), and I wouldn't be surprised if some refactoring down the road silently breaks such an invariant.
My not-so-strong opinion is that, until we migrate our codebase to Dafny, it is better not to rely on contracts. Changing a `== 0` to a `<= 0` seems like a very minor change that would it even more easily to reason locally.
If you prefer not to do that change, at least put a comment recalling the BreakStmt invariant on the else branch.

Reply:
About the variable name, I think it would be better to have it meaning what it does, since It's a public API: breakCount looks instinctively like the number of times the word "break" appears. "level" might thus be more appropriate, but you could as well name it explicitly "SurroundingLoopLevelTarget".

Reply:
Unfortunately, `Contract.Invariant`, `Contract.Requires`, and `Contract.Ensures` contracts are not checked at run time, so it happens they are out of date or that bugs in the code are missed. Still, they often express the intent, and in this case the intent can be checked manually at least back to the constructor. For the particular code in this PR, I added a `Contract.Assert` (which is checked at run time in our `DEBUG` build) at this source location. I also added a comment that describes why I expect the asserted condition to hold.

Reply:
I renamed `BreakCount` to `BreakAndContinueCount`.